src/03-ticket-calculator.js

```diff
@@ -54,7 +54,27 @@ const exampleTicketData = require("../data/tickets");
     calculateTicketPrice(tickets, ticketInfo);
     //> "Entrant type 'kid' cannot be found."
  */
-function calculateTicketPrice(ticketData, ticketInfo) {}
+function calculateTicketPrice(ticketData, ticketInfo) {
+  let ticketPrice = 0;
+  if (!ticketData[ticketInfo.ticketType] || ticketInfo.ticketType === 'extras') {
+    return `Ticket type '${ticketInfo.ticketType}' cannot be found.`;
+  } else if (
+    !ticketData[ticketInfo.ticketType].priceInCents[ticketInfo.entrantType]
+  ) {
+    return `Entrant type '${ticketInfo.entrantType}' cannot be found.`;
+  }
+  ticketPrice +=
+    ticketData[ticketInfo.ticketType].priceInCents[ticketInfo.entrantType];
+  if (ticketInfo.extras) {
+    for (let i = 0; i < ticketInfo.extras; i++) {
+      if (!ticketData.extras[ticketInfo.extras]) {
+        return `Extra type '${ticketInfo.extras}' cannot be found.`;
+      }
+      ticketPrice += ticketData.extras.priceInCents[ticketInfo.entrantType];
+    }
+  }
+  return ticketPrice;
+}
```

Comment:
This is very solid work Aisha.  Work on the final function and ask yourself how our `calculateTicketPrice` function could be used in the next function.  

Additionally - start talking over this code with Silas and Michael!  Its very important not only that we can _make this code_ but we can _quickly_ understand it